UI/modules.hpp.in

```diff
@@ -0,0 +1,13 @@
+#include <string>
+#include <unordered_set>
+
+static const std::unordered_set<std::string> unsafe_modules = {
+#ifdef __APPLE__
+	"obs-browser", // Known to occasionally cause issues on macOS
```

Comment:
I don't think we should remove obs-browser - if it is an "unsafe" module, we should fix that instead. Unfortunately that plugin has become a core functionality of OBS, as we require CEF for too many other things already.

If it is indeed unstable, the 'What's New' functionality should not rely on it and we should use `WKWebKit` on macOS instead. It's one or the other.

Reply:
This was added based on @gxalpha's recommendation, he might be able to elaborate more under which circumstances it causes issues.

Generally I think that we could look into moving at least non-source browser elements to something like https://github.com/webview/webview to use the native webviews on various systems.

Reply:
obs-browser was one example of a module I gave that has caused problems in the past (older versions of OBS on macOS 11 iirc, where people were recommended to just delete their scene collection to fix the problem) when advocating for a much more locked down variant of a safe mode, which would function mostly as a mode to get back out of situations where the current scene collections somehow got completely screwed. Such a mode would have greatly reduced core functionality, with obs-browser being a prime candidate of a module that should not be loaded.

All that said, the kind of Safe Mode this PR proposes is a mode without third-party plugins and scripts, and I'm definitely not in favor of disabling obs-browser there. My communication may have been unclear in that regard.
I do think calling it "Safe Mode" may not be the greatest name for the mode that this PR proposes (operating systems generally get very locked down in their Safe Modes), but that's irrelevant to the current discussion.

Reply:
In that case I'll remove it and update the wording to specifically call out that only third-party plugins and scripting will be disabled.

As mentiond, you're not really supposed to use OBS in this mode, it's so you can export your data or perhaps try to delete a source/filter belonging to a plugin that is causing an issue. And while yes, you can do those things by manually copying or editing the JSON data, your average user is not going to understand that.